src/mqtt/IotaMqttServiceImpl.cc

```diff
@@ -221,3 +222,68 @@ std::string iota::esp::ngsi::IotaMqttServiceImpl::doPublishCB(
 
   return cb_response;
 }
+
+std::string iota::esp::ngsi::IotaMqttServiceImpl::doPublishMultiCB(
+    std::string& apikey, std::string& idDevice,
+    std::vector<std::string>& v_json) {
+  if (apikey == "") {
+    std::ostringstream what;
+    what << "Mandatory field missing: ";
+    what << "[apikey]";
+
+    throw iota::IotaException(what.str(), "", 400);
+  }
+
+  if (idDevice == "") {
+    std::ostringstream what;
+    what << "Mandatory field missing: ";
+    what << "[idDevice]";
+    throw iota::IotaException(what.str(), "", 400);
+  }
+
+  std::string cb_response;
+
+  IOTA_LOG_DEBUG(
+      m_logger,
+      "IotaMqttServiceImpl, doPublishMultiCB... payload: [VECTOR OF MEASURES]"
+          << " apikey: [ " << apikey << "] device: [" << idDevice << "]");
+
+  boost::property_tree::ptree pt_cb;
+
+  try {
+    add_info(pt_cb, apikey);
+    std::string serviceMQTT = pt_cb.get<std::string>("service", "");
+
+    boost::shared_ptr<iota::Device> dev =
+        resthandle_ptr_->get_device(idDevice, serviceMQTT);
+
+    iota::ContextElement ngsi_context_element(idDevice, "", "false");
+
+    ngsi_context_element.set_env_info(pt_cb, dev);
+
+    IOTA_LOG_DEBUG(m_logger,
+                   "MQTTService: Creating entity : ["
+                       << ngsi_context_element.get_string()
+                       << "]");  // call to populate internal fields from cache,
+                                 // or by default, etc... not interesated in
```

Comment:
Hola,
Esta es la entrega realizada y si se considera mejorable ahí tienes la rama. Nunca hemos aplicado tus reglas de estilo ni reglas de revisión, ni vamos a entrar en estas discusiones.

Saludos,
Agustin.


De: Ken Zangelin <notifications@github.com<mailto:notifications@github.com>>
Responder a: telefonicaid/fiware-IoTAgent-Cplusplus <reply@reply.github.com<mailto:reply@reply.github.com>>
Fecha: miércoles, 23 de marzo de 2016, 20:40
Para: telefonicaid/fiware-IoTAgent-Cplusplus <fiware-IoTAgent-Cplusplus@noreply.github.com<mailto:fiware-IoTAgent-Cplusplus@noreply.github.com>>
Asunto: Re: [fiware-IoTAgent-Cplusplus] Feature/mqtt multiattribute (#414)


In src/mqtt/IotaMqttServiceImpl.cc<https://github.com/telefonicaid/fiware-IoTAgent-Cplusplus/pull/414#discussion_r57223717>:

> +  try {
> +    add_info(pt_cb, apikey);
> +    std::string serviceMQTT = pt_cb.get<std::string>("service", "");
> +
> +    boost::shared_ptr<iota::Device> dev =
> +        resthandle_ptr_->get_device(idDevice, serviceMQTT);
> +
> +    iota::ContextElement ngsi_context_element(idDevice, "", "false");
> +
> +    ngsi_context_element.set_env_info(pt_cb, dev);
> +
> +    IOTA_LOG_DEBUG(m_logger,
> +                   "MQTTService: Creating entity : ["
> +                       << ngsi_context_element.get_string()
> +                       << "]");  // call to populate internal fields from cache,
> +                                 // or by default, etc... not interesated in


typo: interesated

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub<https://github.com/telefonicaid/fiware-IoTAgent-Cplusplus/pull/414/files/cfabb5407a210b714c5af47cb9a528438a10d8be#r57223717>

________________________________

Este mensaje y sus adjuntos se dirigen exclusivamente a su destinatario, puede contener información privilegiada o confidencial y es para uso exclusivo de la persona o entidad de destino. Si no es usted. el destinatario indicado, queda notificado de que la lectura, utilización, divulgación y/o copia sin autorización puede estar prohibida en virtud de la legislación vigente. Si ha recibido este mensaje por error, le rogamos que nos lo comunique inmediatamente por esta misma vía y proceda a su destrucción.

The information contained in this transmission is privileged and confidential information intended only for the use of the individual or entity named above. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution or copying of this communication is strictly prohibited. If you have received this transmission in error, do not read it. Please immediately reply to the sender that you have received this communication in error and then delete it.

Esta mensagem e seus anexos se dirigem exclusivamente ao seu destinatário, pode conter informação privilegiada ou confidencial e é para uso exclusivo da pessoa ou entidade de destino. Se não é vossa senhoria o destinatário indicado, fica notificado de que a leitura, utilização, divulgação e/ou cópia sem autorização pode estar proibida em virtude da legislação vigente. Se recebeu esta mensagem por erro, rogamos-lhe que nos o comunique imediatamente por esta mesma via e proceda a sua destruição